app/src/main/java/com/duckduckgo/app/browser/httpauth/db/HttpAuthEntity.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.httpauth.db
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+// CREATE TABLE httpauth (_id INTEGER PRIMARY KEY, host TEXT, realm TEXT, username TEXT, password TEXT, UNIQUE (host, realm) ON CONFLICT REPLACE);
```

Comment:
Question: how probable is it that WebView changes the schema of this database? is this schema defined in their codebase? if so, could we check if they have changed any time in the past?

Reply:
The documentation says webview never reads from this db.
The [source code](https://github.com/chromium/chromium/blame/master/android_webview/java/src/org/chromium/android_webview/HttpAuthDatabase.java) for the creation of the DB has been untouched for 8 years.

I think it's unlikely it'll change however we could always just migrate the entire DB and use our own to ensure the system does not touch it.